identity-wallet/src/state/credentials/mod.rs

```diff
@@ -94,10 +94,11 @@ impl From<CredentialFormats<WithCredential>> for VerifiableCredentialRecord {
                     )
                 };
 
-                let mut issuance_date = String::new();
-                if !credential_display["issuanceDate"].is_null() {
-                    issuance_date = credential_display["issuanceDate"].clone().to_string().replace('\"', "");
-                }
+                let issuance_date = credential_display["issuanceDate"]
+                    .as_str()
+                    .map(ToString::to_string)
+                    .unwrap_or_default()
+                    .replace('\"', "");
```

Comment:
TL;DR: don't need `.replace('\"', "")` here.

this is not necessary, there won't be an instance where you'll encounter `'\"'`.

In your previous solution you needed it because you made use of the `serde_json::Value::to_string()` function which may result in somewhat confusing behavior. It is `serde_json::Value::to_string()` that produces those `'\"'`, so you don't need to worry about them if you don't use that method.

```rust
#[test]
fn test() {
    let credential_display = json!({
        "issuanceDate": "2022-00-00T00:00:00Z"
    });

    // this will result in: `"\"2022-00-00T00:00:00Z\""`
    let a = credential_display["issuanceDate"].to_string(); // <-- a bandaid solution here is to add `.replace('\"', "")` at the end

    // this will result in `"2022-00-00T00:00:00Z"`
    let b = credential_display["issuanceDate"]
        .as_str()
        .map(ToString::to_string)
        .unwrap();

    // So this assertion will fail, because they're not the same.
    assert_eq!(a, b);
}
```